src/main/java/reposense/git/GitClone.java

```diff
@@ -59,4 +60,50 @@ private static void clone(RepoLocation location, String repoName) throws IOExcep
         Files.createDirectories(rootPath);
         runCommand(rootPath, "git clone " + addQuote(location.toString()));
     }
+
+    /**
+     * Spawns a process to clone repo specified in {@code repoConfig}. Does not wait for process to finish executing.
+     * Can only handle a maximum of one spawned process at any time and subsequent calls are ignored if current process
+     * is still running.
+     */
+    public static void spawnCloneProcess(RepoConfiguration repoConfig) throws GitCloneException {
```

Comment:
Let's shift this 2 methods to `RepoCloner` as well, since its only utilized by it. That way we can save `crp` from being a static object as well. 

By the way, if only a maximum of one spawned process should be called at any point of time, we should use Assertion here to prevent multiple calls.